src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,39 +1,67 @@
-import React from 'react';
+import React, { SetStateAction, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [user, setUser] = useState<User>();
```

Comment:
You are initializing the state with a value that comes from a prop. This is not recommended as it can lead to bugs and inconsistencies in the state. The state should be the 'single source of truth'.